cmd/minishift/cmd/start.go

```diff
@@ -276,6 +278,7 @@ func initStartFlags() *flag.FlagSet {
 	startFlagSet.String(startFlags.DiskSize.Name, constants.DefaultDiskSize, "Disk size to allocate to the Minishift VM. Use the format <size><unit>, where unit = b, k, m or g.")
 	startFlagSet.String(startFlags.HostOnlyCIDR.Name, "192.168.99.1/24", "The CIDR to be used for the minishift VM. (Only supported with VirtualBox driver.)")
 	startFlagSet.StringArrayVar(&dockerEnv, "docker-env", nil, "Environment variables to pass to the Docker daemon. Use the format <key>=<value>.")
+	startFlagSet.StringSliceVar(&engineOpt, "docker-opt", nil, "Arbitrary flags to include with Docker daemon in the form flag=value")
```

Comment:
`docker-opt` is not communicating properly about the purpose of the flag. My suggestion would be `docker-opt-flag` i.e.optional flag for  docker daemon